src/main/java/bio/terra/dao/DatasetDao.java

```diff
@@ -185,12 +185,18 @@ private Dataset retrieveWorker(String sql, MapSqlParameterSource params) {
         return datasetSources;
     }
 
-    public List<DatasetSummary> retrieveDatasets(int offset, int limit) {
-        logger.debug("retrieve datasets offset: " + offset + " limit: " + limit);
-        String sql = "SELECT id, name, description, created_date FROM dataset" +
-                " ORDER BY created_date OFFSET :offset LIMIT :limit";
-        MapSqlParameterSource params = new MapSqlParameterSource().addValue("offset", offset)
-                .addValue("limit", limit);
+    public List<DatasetSummary> retrieveDatasets(int offset, int limit, String sort, String direction, String filter) {
+        logger.debug("retrieve datasets offset: " + offset + " limit: " + limit + " sort: " + sort +
+            " direction: " + direction + " filter:" + filter);
+        String where = DaoUtils.whereClause(filter);
+        String sql = "SELECT id, name, description, created_date FROM dataset " + where +
```

Comment:
Not here thankfully, I believe it is going easy on me because the string returned by `whereClause` is a constant string that isn't dangerous on its own. I use a named parameter below to provide the string to filter on which will get escaped properly by the `jdbcTemplate` when it creates the query.